composer.json

```diff
@@ -13,6 +13,7 @@
     "homepage": "https://getrector.com",
     "require": {
         "php": "^8.2",
+        "ext-dom": "*",
```

Comment:
any polyfill for dom extension can be used? if not, fallback when `DOMDocument` class not exists may be needed

Reply:
@samsonasik I don't know.

`PHPUnit` requires both  `ext-xml` and `ext-dom` so i assumed anyone using `PHPUnit` can use `Rector` without any issue.

https://github.com/sebastianbergmann/phpunit/blob/1ff959e4f630995423064d8a5b808c545f7a1a9d/composer.json#L26

Reply:
I see, the scoped composer.json that shipped is under build dir, so probably add at https://github.com/rectorphp/rector-src/blob/main/build/target-repository/composer.json with "suggest" config in case of want to use custom rule command

Reply:
Can we drop this requirement in composer.json and do instead a explicit `extension_loaded` call in this single rule we require it.


That way people using rector, but not this single rule, can still run without ext-dom

Reply:
I think `suggest` config still needed in composer.json, add at target build https://github.com/rectorphp/rector-src/blob/main/build/target-repository/composer.json

Reply:
@samsonasik @staabm good idea. thanks